src/state/routing/types.ts

```diff
@@ -68,20 +67,20 @@ export interface GetQuoteResult {
   routeString: string
 }
 
-export class InterfaceTrade<
+export class ClassicTrade<
   TInput extends Currency,
   TOutput extends Currency,
   TTradeType extends TradeType
 > extends Trade<TInput, TOutput, TTradeType> {
-  gasUseEstimateUSD: CurrencyAmount<Token> | null | undefined
+  gasUseEstimateUSD: string | null | undefined
   blockNumber: string | null | undefined
 
   constructor({
     gasUseEstimateUSD,
     blockNumber,
     ...routes
   }: {
-    gasUseEstimateUSD?: CurrencyAmount<Token> | undefined | null
+    gasUseEstimateUSD?: string | undefined | null
     blockNumber?: string | null | undefined
     v2Routes: {
       routev2: V2Route<TInput, TOutput>
```

Comment:
nit: I don't think we should re-define these types here. They are already defined by the `Trade` type - that's why it's generic and accepts those three parameters. Instead, we should have a type like this:
```
{ gasUseEstimateUSD?: string | null, blockNumber?: string | null } & TradeConstructorArgs }
```
that we can extract with something like `ConstructorParameters` -> https://www.typescriptlang.org/docs/handbook/utility-types.html#constructorparameterstype